# Project 4 -- Dwijen Chawra

Collaborators:
- Alexiy Buynitsky - called me a monkey and helped me with optimizing sorts

# Question 1

In [1]:
%%bash
# head -n 1 /anvil/projects/tdm/data/stackoverflow/unprocessed/2011.csv
head -n 1 /anvil/projects/tdm/data/stackoverflow/unprocessed/2011.csv | tr "," "\n" | wc -l


head -n 1 /anvil/projects/tdm/data/stackoverflow/unprocessed/2011.csv | tr "," "\n" | head -n 100

191221
What Country or Region do you live in?
Which US State or Territory do you live in?
How old are you?
How many years of IT/Programming experience do you have?
How would you best describe the industry you work in?
Which best describes the size of your company?
Which of the following best describes your occupation?
How likely is it that a recommendation you make will be acted upon?
What is your involvement in purchasing? You can choose more than 1.






What types of purchases are you involved in?





"What is your budget for outside expenditures (hardware
 software
 consulting
 etc) for 2011?"







What type of project are you developing?
Which languages are you proficient in?












What operating system do you use the most?
Please rate your job/career satisfaction
"Including bonus
 what is your annual compensation in USD?"
Which technology products do you own? (You can choose more than one)
















"In the last 12 months
 how much money have you spent on personal

I notice that `tr` was unable to properly replace the comma with newlines, and `wc -l` still returned 191221 for the line count because it replaced the ends of the rows with newlines.

`tr` was able to properly display and replace the commas with new lines, and the line count was more accurate.

# Question 2

In [2]:
%%bash
head -n 1 /anvil/projects/tdm/data/stackoverflow/processed/2011.csv | tr ";" "\n" | wc -l

head -n 1 /anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales_cleaner.csv | tr ";" "\n" | wc -l

19
24


`2011.csv` has 19 columns.

`iowa_liquor_sales_cleaner.csv` has 24 columns.

# Question 3

### Bottles sold:

In [5]:
%%bash
cut -d ";" -f 21 /anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales_cleaner.csv | sort -n | tail -n 5


5280
6600
11124
15000
15000


### Gallons sold:

In [6]:
cut -d ";" -f 24 /anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales_cleaner.csv | sort -n | tail -n 5

SyntaxError: invalid syntax (268557555.py, line 1)

All the 5 largest orders are from the "Hy-Vee #3 / BDI / Des Moines" store.

# Question 4

In [1]:
%%bash

cat /anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales_cleaner.csv | cut -d ";" -f 18 | sort -n | uniq -c

      1 
      1 $144.00
     14 0
      1 Bottle Volume (ml)
      1 12
      1 15
   3906 50
  39299 100
    592 150
 440717 200
    727 250
  61639 300
    130 301
1224058 375
   1004 400
 523506 500
    349 502
 141093 600
    187 603
5719504 750
  13723 800
   1235 850
    708 900
    427 950
1787513 1000
    303 1125
  13949 1200
    112 1500
      3 1508
2550403 1750
     29 1800
    762 1850
    516 1950
     23 2000
    143 2250
   4600 2400
     42 2550
  58173 3000
    301 3500
    161 3600
     29 3900
     49 4500
    476 4800
    603 6000
      1 7580
      3 9000
     24 31500
      8 140000
      3 180000
      2 189000
      1 225000
     24 378000


Ignoring the top of the output above, there is a distribution of various bottle sizes in the output. 

# Question 5

In [2]:
%%bash
cat /anvil/projects/tdm/data/iowa_liquor_sales/iowa_liquor_sales_cleaner.csv | cut -d ";" -f 20 | cut -c 2 | sort | uniq -c | tail -n 11 | head -n 10

   3090 0
5958386 1
1966706 2
 578190 3
 622197 4
 702561 5
 643977 6
 740031 7
 550582 8
 825346 9


This distribution appears to be following the distribution, only up until 2 and 3. Everything past that appears to be rising a little, indicating that the stores might be pricing many items starting with 1 to make them appear smaller, and then similarly towards 8 and 9 so that those liquors are one digit smaller than the next order of 10 (EX: $80 vs $100, though $80 starts with a higher digit, it has less digits total.)